internal/vsp/vsp.go

```diff
@@ -59,7 +61,7 @@ type Config struct {
 	Params *chaincfg.Params
 }
 
-func New(cfg Config) (*Client, error) {
+func New(cfg Config, log slog.Logger) (*Client, error) {
```

Comment:
do you think it would make more sense to provide the logger in the config? and if it's nil, either don't log anything (probably the best default) or have some default logger to stdout or something (i'm less thrilled by this idea)

Reply:
I added it here to avoid adding it in every place where a `vsp.Config` instance is created (there are a lot of them - `server.go` is peppered).

Just to explain my reasoning...

Personally I don't view a logger as logically being a config item. I view config as a set of static values which tweak/toggle various behaviours of the component they are passed in to. The values in config may be mandatory or they may have defaults.

A logger is an independent component which implements its own behaviour (and probably has its own config internally). I put logger in the same "not config" category as database/RPC connections, caches, clocks, random number generators. Distinct from config, I'd consider these to be non-optional. They are typically fundamental things your component needs to do its job.

That said, this is all pretty subjective. I'm happy to change it if you like, just let me know what your preferred solution is. 

Perhaps we could default to `slog.Disabled` if you want logger to be optional?

Reply:
it's fine as is, just wanted to run the idea past you for your opinion